src/kernel.ts

```diff
@@ -154,7 +161,8 @@ class Kernel {
    *
    * Get the list of running kernels.
    */
-  static list(kernelServiceUrl: string): Promise<IKernelId[]> {
+  static list(baseUrl: string): Promise<IKernelId[]> {
+    var kernelServiceUrl = utils.urlJoinEncode(baseUrl, KERNEL_SERVICE_URL)
     return utils.ajaxRequest(kernelServiceUrl, {
```

Comment:
Should we be validating the ajax responses? If the server returns something invalid, the return type would be a lie.